### PR TITLE
#352 micro skeleton DONE

### LINKED ISSUE #352: [5] micro-skeleton

### PR DESCRIPTION
**Pull request by MichalJagodzinski**

### CODE REVIEW

File: elibrary-micro/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    depends_on:
+      - dbpostgresql
+  dbpostgresql:
+    image: postgres
+    ports:
+      - "5432:5432"
+    environment:
+      - POSTGRES_PASSWORD=password
+      - POSTGRES_USER=postgres
+      - POSTGRES_DB=jfs-operations
```
Comment (JaRutkowski) [inline elibrary-micro/deployment.yml]: Please change `POSTGRES_DB` to `elibrary-micro`.

File: elibrary-micro/deployment.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: jfs-operations-app
```
Comment (JaRutkowski) [inline elibrary-micro/deployment.yml]: Please change `name` to `elibrary-micro`.

File: elibrary-micro/deployment.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+spec:
+  selector:
+    matchLabels:
+      app: jfs-operations-app
```
Comment (JaRutkowski) [inline elibrary-micro/deployment.yml]: Please change `app` to `elibrary-micro`.

File: elibrary-micro/deployment.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  replicas: 5
+  template:
+    metadata:
+      labels:
+        app: jfs-operations-app
```
Comment (JaRutkowski) [inline elibrary-micro/deployment.yml]: Please change `app` to `elibrary-micro`.

File: elibrary-micro/deployment.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    spec:
+      containers:
+        - name: jfs-operations-app
```
Comment (JaRutkowski) [inline elibrary-micro/deployment.yml]: Please change `name` to `elibrary-micro`.

File: elibrary-micro/deployment.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+          image: jfs-operations-app
```
Comment (JaRutkowski) [inline elibrary-micro/deployment.yml]: Please change `image` to `elibrary-micro`.

File: elibrary-micro/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    depends_on:
+      - dbpostgresql
+  dbpostgresql:
+    image: postgres
+    ports:
+      - "5432:5432"
+    environment:
+      - POSTGRES_PASSWORD=password
+      - POSTGRES_USER=postgres
+      - POSTGRES_DB=jfs-operations
```
Comment (JaRutkowski) [inline elibrary-micro/docker-compose.yml:18]: Please change `POSTGRES_DB` to `elibrary-micro`.

File: elibrary-micro/pom.xml [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+						</exclude>
+					</excludes>
+				</configuration>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.tomcat.maven</groupId>
+				<artifactId>tomcat7-maven-plugin</artifactId>
+				<version>2.2</version>
+				<configuration>
+					<url>http://jFeeSoft:8080/manager/text</url>
```
Comment (JaRutkowski) [inline elibrary-micro/pom.xml]: Please change `jFeeSoft` to `<todo>`.

File: elibrary-micro/src/main/java/com/javafee/elibrary/micro/config/CommandRunnerConfig.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+	@Value("${spring.profiles.active}")
+	private String activeProfile;
+
+	@Override
+	public void run(String... args) throws Exception {
+		logDBConnectionInfo();
+	}
+
+	private void logDBConnectionInfo() throws Exception {
+		log.info("jfs-operations startup, connected to " + connectionString + " as "
```
Comment (JaRutkowski) [inline elibrary-micro/src/main/java/com/javafee/elibrary/micro/config/CommandRunnerConfig.java]: Please change `jfs-operations` to `elibrary-micro`.

File: elibrary-micro/src/main/java/com/javafee/elibrary/micro/infrastructure/exception/CustomGlobalExceptionHandler.java [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+					.stream()
+					.map(DefaultMessageSourceResolvable::getDefaultMessage)
+					.collect(Collectors.toList());
+		else
+			errors.add(e.getMessage());
+		body.put("errors", errors);
+		return body;
+	}
+
+	private JfsExceptionDto prepareCustomExceptionDto(Exception e, HttpStatus status, List<String> messages, String path, String method) {
```
Comment (JaRutkowski) [inline elibrary-micro/src/main/java/com/javafee/elibrary/micro/infrastructure/exception/CustomGlobalExceptionHandler.java]: Please change the object name from `JfsExceptionDto` to `ElibraryMicroExceptionDto`.

File: elibrary-micro/src/main/resources/application-prod.properties [hunk 11]
```diff
@@ -0,0 +1,27 @@
+# Database config
+spring.datasource.url                     = jdbc:postgresql://jFeeSoft/jfs-operations
```
Comment (JaRutkowski) [inline elibrary-micro/src/main/resources/application-prod.properties]: Please change `jfs-operations` to `elibrary-micro`.

File: elibrary-micro/src/main/resources/application-prod.properties [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+spring.datasource.driver-class-name       = org.postgresql.Driver
+spring.datasource.username                = jfs-operations
```
Comment (JaRutkowski) [inline elibrary-micro/src/main/resources/application-prod.properties]: Please change `jfs-operations` to `elibrary-micro`.

File: elibrary-micro/src/main/resources/logback-spring.xml [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+	<logger name="com.jfs.operations" level="DEBUG" additivity="false">
+		<appender-ref ref="FILE"></appender-ref>
+	</logger>
+
+	<logger name="com.jfs.operations" level="INFO" additivity="false">
+		<appender-ref ref="STDOUT"></appender-ref>
+	</logger>
+
+	<logger name="com.jfs.operations.infrastructure" level="INFO" additivity="false">
```
Comment (JaRutkowski) [inline elibrary-micro/src/main/resources/logback-spring.xml]: Please adjust the packages to the proper ones.

**Review by JaRutkowski — CHANGES_REQUESTED** [review]
Please see all the comments.

File: elibrary-micro/deployment.yml (hunk 1 above)
Comment (MichalJagodzinski) [reply elibrary-micro/deployment.yml]: done

File: elibrary-micro/deployment.yml (hunk 2 above)
Comment (MichalJagodzinski) [reply elibrary-micro/deployment.yml]: done

File: elibrary-micro/deployment.yml (hunk 3 above)
Comment (MichalJagodzinski) [reply elibrary-micro/deployment.yml]: done

File: elibrary-micro/deployment.yml (hunk 4 above)
Comment (MichalJagodzinski) [reply elibrary-micro/deployment.yml]: done

File: elibrary-micro/deployment.yml (hunk 5 above)
Comment (MichalJagodzinski) [reply elibrary-micro/deployment.yml]: done

File: elibrary-micro/deployment.yml (hunk 6 above)
Comment (MichalJagodzinski) [reply elibrary-micro/deployment.yml]: done

File: elibrary-micro/docker-compose.yml (hunk 7 above)
Comment (MichalJagodzinski) [reply elibrary-micro/docker-compose.yml:18]: done

File: elibrary-micro/pom.xml (hunk 8 above)
Comment (MichalJagodzinski) [reply elibrary-micro/pom.xml]: done

File: elibrary-micro/src/main/java/com/javafee/elibrary/micro/config/CommandRunnerConfig.java (hunk 9 above)
Comment (MichalJagodzinski) [reply elibrary-micro/src/main/java/com/javafee/elibrary/micro/config/CommandRunnerConfig.java]: done

File: elibrary-micro/src/main/java/com/javafee/elibrary/micro/infrastructure/exception/CustomGlobalExceptionHandler.java (hunk 10 above)
Comment (MichalJagodzinski) [reply elibrary-micro/src/main/java/com/javafee/elibrary/micro/infrastructure/exception/CustomGlobalExceptionHandler.java]: done

File: elibrary-micro/src/main/resources/application-prod.properties (hunk 11 above)
Comment (MichalJagodzinski) [reply elibrary-micro/src/main/resources/application-prod.properties]: done

File: elibrary-micro/src/main/resources/application-prod.properties (hunk 12 above)
Comment (MichalJagodzinski) [reply elibrary-micro/src/main/resources/application-prod.properties]: done

File: elibrary-micro/src/main/resources/logback-spring.xml (hunk 13 above)
Comment (MichalJagodzinski) [reply elibrary-micro/src/main/resources/logback-spring.xml]: done

File: elibrary-micro/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,6 @@
+FROM openjdk:17-jdk-alpine
+MAINTAINER experto.com
+VOLUME /tmp
+EXPOSE 8080
+ADD target/jfs-operations-0.0.1-SNAPSHOT.war jfs-operations.war
```
Comment (JaRutkowski) [inline elibrary-micro/Dockerfile]: Please fix the name of the war.

File: elibrary-micro/README.md [hunk 15]
```diff
@@ -0,0 +1,140 @@
+# jfs-operations
```
Comment (JaRutkowski) [inline elibrary-micro/README.md]: Please fix naming in the README.

**Review by JaRutkowski — CHANGES_REQUESTED** [review]

File: elibrary-micro/Dockerfile (hunk 14 above)
Comment (MichalJagodzinski) [reply elibrary-micro/Dockerfile]: done

File: elibrary-micro/README.md (hunk 15 above)
Comment (MichalJagodzinski) [reply elibrary-micro/README.md]: done

**Review by JaRutkowski — APPROVED** [review]

**Pull request merged**